lib/http/timeout/per_operation.rb

```diff
@@ -11,6 +11,40 @@ class PerOperation < Null
       WRITE_TIMEOUT = 0.25
       READ_TIMEOUT = 0.25
 
+      SETTINGS = Set.new(%i[read_timeout write_timeout connect_timeout])
+
+      class << self
+        def parse_options(options)
+          options = options.dup.then { |opts| expand_names(opts) }
+          options.each do |key, value|
+            unless SETTINGS.member?(key) && value.is_a?(Numeric)
+              raise ArgumentError, "invalid option #{key.inspect}, must be numeric " \
+                                   "`.timeout(connect: x, write: y, read: z)`."
+            end
+          end
+
+          raise ArgumentError, "at least one option" if options.empty?
+
+          options
```

Comment:
I believe we also should fail if given options are mistyped or totally wrong ones